### ISSUE TITLE
AI and accessibility: opportunities and risks

### ISSUE THREAD

**Issue by FabienGandon** [w3c/webai#14]
Map scenarios where AI improves or harms accessibility, from assistive tools to automated decision barriers.

**Comment by LJWatson** [comment]
A few initial thoughts... The data most models are trained on is a risk. Models trained on [common crawl](https://commoncrawl.org/) data have been taught using a web that still [struggles with accessibility](https://webaim.org/projects/million/) and consequently, the accessibility mistakes now being made by chatbots and agents bear a striking resemblance to the mistakes made by designers and developers.

There may be an opportunity to counter this risk, but it needs more exploration. Given better training data, robust skills, clear guardrails etc., might it be possible for AI agents to assist designers and developers more reliably in the creation of accessible interfaces?

Another risk with the same cause is the representation of People With Disabilities (PWD). The training data includes all the misinformation, misconceptions, and mischaracterisations of PWD that has accumulated over time. For example, received wisdom has it that most blind people read Braille and/or have a service animal, when in fact both of those groups are small minorities within the blind/low vision community.

An opportunity that is already evident is the use of computer vision/image recognition in AI models. Browsers have had the ability to request a description of images without alternative textsfor some time, and more recently screen readers have introduced the ability for users to query popular AI models for text descriptions (with the added benefit of being able to ask follow-up questions to derive more specific or detailed information).

Another evident opportunity is the ability to access content from within large, complex, or inaccessible documents. Until the advent of recent AI models, the only alternative to an inaccessible PDF for a screen reader user was to use Optical Character Recognition (OCR) to extract the text content. This is a useful solution but it's a "blunt instrument". Now it's possible to upload the document to a chatbot and prompt it to extract the information you need in the form or format you need it.

Similarly, the ability to use AI models to summarise complex documents can be helpful to neurodiverse people who find it difficult to consume or process large amounts of content or interpret abstract concepts. For people who find reading difficult, the conversational capabilities available in many chatbots can lower the barriers.

The risk of hallucinations is, of course, ever present, but that is not in and of itself an accessibility risk.

**Comment by AutoSponge** [comment]
## Opportunities

Academic studies have shown where LLMs have better search/retrieval capability when the source documents are well structured, markdown being preferred. Between this fact and the cost of context (measured in tokens), several browser automation projects have popped up to give agents better access to the web and all picked the axTree as the primary resource. When saved/cached, it can get converted to markdown. Edge providers, like Cloudflare, have created services to offload this repetitive task. Sites that do this well, get ranked higher in "SEO" when AI is layered on top. If only 3 citations appear at the top of Google, first page is no longer good enough.

All this to say, accessibility-first (which we've been hammering on for years) is also agent-first. They intersect on wanting well-formed, structured information in plain language. Robots are people too?

## Risks

- If anyone can vibe code a UI, but the agents pick the shortest (lazy) path, it won't be accessible. 
- UIs moving to the terminal (because agents like the CLI), causes more accessibility issues than web.
- Social engineering attacks are way more sophisticated. The speed at which someone can be tricked into divulging PII thinking they're talking to a trusted person OR a trusted agent is astounding. Some attacks even [target the agent](https://www.microsoft.com/en-us/security/blog/2026/02/10/ai-recommendation-poisoning/).

**Comment by mgifford** [comment]
I am experimenting. I've shared some of these on Slack, but:

What if we defined our expectations, much like a AGENTS.md file
  https://mgifford.github.io/ACCESSIBILITY.md/

I've found this works reasonably well, when for instance I want to implement dark mode, and want to point to an example like:
  https://github.com/mgifford/ACCESSIBILITY.md/blob/main/examples/LIGHT_DARK_MODE_ACCESSIBILITY_BEST_PRACTICES.md

Why don't we give the bots a list of those we trust (and those who are willing to have their sites crawled):
  https://github.com/mgifford/ACCESSIBILITY.md/blob/main/examples/TRUSTED_SOURCES.yaml

I got this idea from Ted Drake: 
  https://github.com/7mary4/a11y-ai-training

If you're logged into Google in your browser, try going to the and adding a URL:
https://gemini.google.com/gem/1-oRtNsS6CLM1WMI3-JJ9bn_Se99us5sJ

I've definitely got more ways that this can be an opportunity for our industry. There are obviously a lot of risks. Lots of certainty too in terms of the dangers of data centers. This is a very sharp double-edged sword.

**Comment by FabienGandon** [comment]
Thanks for the previous inputs.
Here is just a pointer to the talk of Matthew Atkinson during the IG call:
https://www.w3.org/2026/02/09-webai-minutes.html#92d6

**Comment by AutoSponge** [comment]
I've been following @mgifford experiments trying to guide AI. I also think Michael Fairchild's https://github.com/microsoft/a11y-llm-eval is important for evaluating the "code author." But I've recently been aware of the problem faced in large code bases where AI's ability to pass tests or move quickly explodes into an unmaintainable mess. We used to call this spaghetti code. Every engineer had their hydra bugs back in the day, then they learned to be more disciplined. So, the problem here compounds. Initially, AI writes code that scores low on accessibility. Allowed to continue, it flourishes into a full-blown disaster. AI only magnifies what's already happening, adding velocity. WebAIM's research shows what's happening. Now multiply it, reduce the support staff to increase profits, and watch the lawsuits roll in.

Several years ago I suggested an addition to well-known urls, `/accessibility-issue`. The idea is that anyone could easily direct the UA or an extension to submit a detailed technical issue on their behalf to a POST-only endpoint. The endpoint would only need to dedupe and guard against DDoS. Now that AI and AI-based UAs are emerging as user personas, it may be time to dust off that proposal again.

**Comment by abbott567** [comment]
## Opportunities 

Just weighing in with a few off the top of my head from my work with LLMs. Worth noting though that the opportunities are pretty dependent on which LLM you use, as I find they all have different strengths and weaknesses.

1. Converting inaccessible PDFs into more accessible formats like Markdown:
   - I've had good results using Claude to do this
   - However, it does involve a particular process which takes longer and burns through more tokens

2. Helping to decipher and understand regulations, legislation and policy:
   - I've had good results using Perplexity to do this
   - You can restrict it to only use specific sources
   - It's better than most when it comes to it's citations, so you can sense-check exactly where it got the information etc

3. Advancing automated tooling to do more of the human dependent stuff around language etc (with the heavy caveat that it needs human reviewed) for example:
   - I've had good results using Gemini and ChatGPT for this
   - Links / labels for purpose, clarity, context, intent
   - Headings for purpose, clarity, context
   - Image visuals against alt text
   - Fig captions / summaries against visual charts
   - Error messages for purpose, clarity

4. Information architecture, sorting and categorising massive amounts of documentation
  - I've had good results using Claude for this
  - However, like a human it often struggles when things don't fit nicely into a single category

## Risks

1. They've been trained on so much bad code that they will confidently write terrible code, and then tell you it's accessible and follows best practice etc, which I think will create a false sense of security for many organisations / developers

2. A lot of people don't know how to do "spec driven prompting" or manage context windows, which massively affects the outputs and accuracy from agentic AI when doing complex tasks

3. People get lazy and will not properly sense check what the AI is doing, especially if it's churning out a weeks worth of work in a day
